packages/react/src/exampleTypes/ExpandableTagsQuery/index.js

```diff
@@ -79,73 +195,114 @@ let TagsWrapper = observer(
+              removeTag={(tag) => {
+                tree.mutate(node.path, {
+                  tags: _.reject({ [tagValueField]: tag }, node.tags),
+                })
+              }}
+              tagStyle={getTagStyle(node, tagValueField)}
+              submit={tree.triggerUpdate}
+              Tag={TagWithPopover}
+              style={{ flex: 1, border: 0 }}
+              {...props}
+            />
+          </GridItem>
+          <GridItem place="center">
+            <TextButton
+              style={
+                // Show suggestion lightbulb if min of 3 non numeric tags exist,
+                // including numbers ups the chance of producing bad suggestions
+                generationTagInputs(node.tags)?.length > 2 &&
+                enableKeywordGenerations
+                  ? { width: 35 }
+                  : { display: 'none' }
               }
-              tree.mutate(node.path, { tags })
-              onAddTag(tags)
-            }}
-            removeTag={(tag) => {
-              tree.mutate(node.path, {
-                tags: _.reject({ [tagValueField]: tag }, node.tags),
-              })
-            }}
-            tagStyle={getTagStyle(node, tagValueField)}
-            submit={tree.triggerUpdate}
-            Tag={TagWithPopover}
-            style={{ flex: 1, border: 0 }}
-            {...props}
-          />
-        </GridItem>
-        <GridItem place="center">
-          <Popover
-            style={{ width: 'auto' }}
-            position={popoverPosition}
-            arrow={popoverArrow}
-            offsetY={popoverOffsetY}
-            closeOnPopoverClick={false}
-            trigger={
-              <div>
-                <Icon icon="TableColumnMenu" />
-              </div>
-            }
-          >
-            {(close) => (
-              <ActionsMenu
-                {...{
-                  node,
-                  tree,
-                  close,
-                  actionWrapper,
-                  joinOptions,
-                }}
-              />
-            )}
-          </Popover>
-        </GridItem>
-      </Grid>
+              onClick={async () => {
+                // Ensure first time user clicks the lightbulb words generate.
+                // Regenerate words and reset existing for clean loading experience
+                // on subsequent clicks.
+                console.log('generationsCollapse', generationsCollapse, node)
+                if (F.view(generationsCollapse)) {
+                  F.off(generationsCollapse)()
+                  if (
+                    !node.context.keywordGenerations &&
+                    !node.generateKeywords
+                  ) {
+                    await tree.mutate(node.path, { generateKeywords: true })
+                    // Await on the first generation to update false,because
+                    // subsequent button clicks(hide/show) can have edge cases
+                    tree.mutate(node.path, { generateKeywords: false })
```

Comment:
let's factor this out to a method since it's in here twice and the logic is not obvious